doc/manual/src/glossary.md

```diff
@@ -271,6 +271,21 @@
 
   The epsilon symbol. In the context of a package, this means the version is empty. More precisely, the derivation does not have a version attribute.
 
+- [package]{#package}
+
+  1. A software package; typically a collection of programs, files and data.
```

Comment:
```suggestion
  1. A software package; typically a collection of executable files and other data.
```

Everything is a file around here.

Reply:
What I'm really trying to do here is to explicitly not redefine this established term from outside our ecosystem, or even UNIX. That way we avoid a little bit of confusion, or perhaps people pointing at this definition saying in Nix land we must use THIS definition of package, or you're WRONG.
Hence programs instead of executables, as executables in the context of Nix would be associated with the executable files in NAR. This is not that.
Coincidentally, it also allows outputs and `.meta` to fit within this definition. That makes meaning 2 not an overload but a specialisation, which I think is great.

Readers don't need to be distracted by all that reasoning but maybe a comment for editors is helpful:

```suggestion
  <!-- @editors: The goal of this definition is to _not_ redefine "package". Hence the vague definition which is not aligned with the file system object terminology. That's intentional. We're interfacing with the world through "package", so let's not be weird. -->
  1. A software package; typically a collection of programs, files and data.
```

Wdyt?

Reply:
> Hence programs instead of executables, as executables in the context of Nix would be associated with the executable files in NAR. This is not that.

I don't understand. How is a program not an executable file? What does this have to do with NARs?

A program implies being runnable. Software doesn't have to be that, and the longer I think about it, it seems like a historical buzzword for data.

Does that work for you?

```suggestion
  1. A software package; a collection of files and other data.
```

Reply:
I've picked your last suggestion. I like that it is appropriately vague, and its awkwardness of implying that the files aren't the only data is perfect.
The existence of something remotely instruction-like feels important for it to be software, but we already mentioned software, so that's fine.